BlueShift-iOS-SDK/BlueShiftHttpRequestBatchUpload.m

```diff
@@ -51,10 +51,14 @@ + (void)createBatches {
                             if(requestOperation.parameters != nil) {
                                 [paramsArray addObject:requestOperation.parameters];
                             }
-                            [context deleteObject:operationEntityToBeExecuted];
                             NSError *saveError = nil;
                             @try {
-                                [context save:&saveError];
+                                if(context && [context isKindOfClass:[NSManagedObjectContext class]]) {
+                                    [context deleteObject:operationEntityToBeExecuted];
+                                }
+                                if(context && [context isKindOfClass:[NSManagedObjectContext class]]) {
```

Comment:
Saving context checking it's validity was a hot fix. Ideally , any context being saved in background should be encapsulated in performBlock: or performBlockAndWait: , performBlock: incase is preferred as it creates it's own autorelease pool(solves or bad_excess) and is async (ideal here). 

For cases where fetch return some data , i prefer passing a completion block with invocation.